graylog2-server/src/main/java/org/graylog2/migrations/RoleRemover.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.migrations;
+
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.Updates;
+import com.mongodb.client.result.DeleteResult;
+import com.mongodb.client.result.UpdateResult;
+import jakarta.inject.Inject;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+import org.bson.types.ObjectId;
+import org.graylog2.database.MongoConnection;
+import org.graylog2.shared.utilities.StringUtils;
+import org.graylog2.users.RoleServiceImpl;
+import org.graylog2.users.UserImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Locale;
+
+import static com.mongodb.client.model.Projections.include;
+
+/**
+ * Removes a role.
+ * Bypasses all the checks, removes even read-only roles.
+ * Because of that it is placed in migrations module, instead of being added to {@link org.graylog2.users.RoleService}
+ */
+public class RoleRemover {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RoleRemover.class);
+
+    private final MongoConnection mongoConnection;
+
+    @Inject
+    public RoleRemover(final MongoConnection mongoConnection) {
+        this.mongoConnection = mongoConnection;
+    }
+
+    public void removeBuiltinRole(final String roleName) {
```

Comment:
Shouldn't this be in the `MigrationHelpers` class?

Reply:
What's wrong with it being in a separate, dedicated class?

Reply:
For discovery: As someone who would be looking for a method to remove roles in the future, I would look in the `MigrationHelpers` class and implement a (duplicate) myself if it's not in there.

Reply:
I see.
I have commited a change where RoleRemover is not used directly, but via MigrationHelpers. It should solve the discovery problem.